src/components/LoadingComponent.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+    <div class="loading-component">
+      <transition name="fade">
+        <div class="page-overlay" v-if="!simple && showSpinner">
+          <div class="progress-wrapper">
+            <div class="progress-indicator progress-red"></div>
+            <div class="progress-txt">Laden...</div>
+          </div>
+        </div>
+        <div class="progress-indicator progress-red" v-if="simple"></div>
+      </transition>
+    </div>
+</template>
+
+<script>
+import { HTTPStatus } from '../services/datareader'
+
+export default {
+  name: 'LoadingComponent',
+  components: {
+  },
+  props: {
+    simple: {
+      default: false,
+      type: Boolean
+    }
+  },
+  data () {
+    return {
+      HTTPStatus,
+      showSpinner: false
+    }
+  },
+  methods: {
+    shouldShowSpinner (pending) {
+      if (pending > 0) {
+        clearTimeout(this.spinnerHideTimeout)
```

Comment:
spinnerHideTimeout is not defined/initialized in data
Should also be prevented against clearTimeout on no value (rare case, but if the service is already loading when the component is created, pending gets 0 without ever having been positive)

Reply:
cleartimeout can be called with a non-existing variable, so this is no problem